dissolve-struct-lib/src/main/scala/ch/ethz/dalab/dissolve/optimization/SolverUtils.scala

```diff
@@ -159,12 +160,13 @@ object SolverUtils {
     val featureFn = dissolveFunctions.featureFn _
     val oracleFn = dissolveFunctions.oracleFn _
     val lossFn = dissolveFunctions.lossFn _
+    val classWeight = dissolveFunctions.classWeights _ 
 
     var hingeLosses: Double = 0.0
     for (i <- 0 until data.size) {
       val yStar_i = oracleFn(model, data(i).pattern, data(i).label)
-      val loss_i = lossFn(yStar_i, data(i).label)
-      val psi_i = featureFn(data(i).pattern, data(i).label) - featureFn(data(i).pattern, yStar_i)
+      val loss_i = lossFn(yStar_i, data(i).label)//*classWeight(data(i).label)
+      val psi_i = featureFn(data(i).pattern, data(i).label) - featureFn(data(i).pattern, yStar_i)//*classWeight(data(i).label)
 
```

Comment:
need to be sure if hinge-loss changes as well or not. also check if at prediction time everything is ok.
important quantities: primal objective, dual objective, gap, error (#misclassifications, confusion matrix).